java/src/main/java/com/genexus/db/driver/GXDBMSdameng.java

```diff
@@ -0,0 +1,246 @@
+	{
+		return CommonUtil.ymdhmsToT_noYL(1, 1, 1, 0, 0, 0);
+	}
+
+	public boolean useDateTimeInDate()
+	{
+		return false;
+	}
+	public boolean useCharInDate()
+	{
+		return false;
+	}
+	
+
+	public void setConnectionProperties(java.util.Properties props)
+	{
+		props.put("fixedString", "true");
+	}
+
+	public void onConnection(GXConnection con)
+	{
+	}
+	
+	public java.util.Date serverDateTime(GXConnection con) throws SQLException
+	{
+		ResultSet rslt = con.getStatement("_ServerDT_", "SELECT LOCALTIMESTAMP()", false).executeQuery();
+		
+		rslt.next();
+		Date value = rslt.getTimestamp(1);
+		rslt.close();
+
+		return value;
+	}
+	
+	public String serverVersion(GXConnection con) throws SQLException
+	{
+		return "";
+	}	
+	
+	public String connectionPhysicalId(GXConnection con)
+	{
+		try
+		{
+			
+			ResultSet rslt = con.getStatement("_ConnectionID_", "SELECT SESSID() AS S", false).executeQuery();
+		
+			rslt.next();
+			int value = rslt.getInt(1);
+			rslt.close();
+
+			return String.valueOf(value);
+		}
+		catch (SQLException e)
+		{
+			return "";
+		}
+	}
+
+	public boolean getSupportsAutocommit()
+	{
+		return true;
+	}
+
+	public void commit(Connection con) throws SQLException
+	{
+		//Las reorgs corren en modo autocommit con lo cual no se debe hacer commit ni rollback implicitos.
+		if	(!com.genexus.ApplicationContext.getInstance().getReorganization())
+		{
+			con.commit();
+		}
+	}
+
+	public void rollback(Connection con) throws SQLException
+	{
+		//Las reorgs corren en modo autocommit con lo cual no se debe hacer commit ni rollback implicitos.
+		if	(!com.genexus.ApplicationContext.getInstance().getReorganization())
+		{		
+			con.rollback();
+		}
+	}
+	public boolean ignoreConnectionError(SQLException e)
+	{
+		return false;
+	}
+        
+        public boolean rePrepareStatement(SQLException e)
+        {
+		return false;
+	}
+
+	public boolean getSupportsQueryTimeout()
+	{
+		return true;
+
+	}
+
+	public boolean useStreamsInNullLongVarchar()
+	{
+		return false;
+	}
+
+	public boolean useStreamsInLongVarchar()
+	{
+		return true;
+	}
+
+	private static java.lang.reflect.Method PUTBYTES;
+	
+	/** Setea los datos de un blob
+	 * El parametro blob debe ser una instancia de java.sql.Blob o descendiente
+	 * Esta puesto como Object para que compile sin problemas en JSharp
+	 */
+	public void setBlobData(Object blob, InputStream stream, int length)throws Exception
+	{
+		try
+		{
+			byte [] bytes = new byte[length];
+			com.genexus.PrivateUtilities.readFully(stream, bytes, 0, length);
+			if(PUTBYTES == null)
+			{
+				PUTBYTES = Class.forName("java.sql.Blob").getMethod("setBytes", new Class[]{long.class, byte[].class});
+			}
+			PUTBYTES.invoke(blob, new Object[]{new Long(1), bytes});
+		}catch(Exception e)
+		{
+				System.err.println(e.toString());
+				throw e;
+		}
+	}
+	
+	public int getId()
+	{
+		return DBMS_ORACLE;
+	}
+        public int getLockRetryCount(int lockRetryCount, int waitRecord){
```

Comment:
Just a (tiny) minor suggestion: There seems to be some indentation issues in the source code. Maybe we can re-format the source code with the IDE . As this is a new java class, I think it would be better to have the identation correctly at first commit.